### PR TITLE
Improve last_modified performances

### LINKED ISSUE #933: performance of last_modified
hello,

in a scenario with ~2500 commits, where each commit *N* created two files (`foo/a`*N* and `foo/b`*N* -- i.e. all in the same directory), when I iterate over the directory and call `last_modified <store> "foo/a*N*"`, I see a massive amount of computation (99%) done in `last_modified` (pls find attached landmarks output), leading to an overall processing time of ~76 seconds (which is not acceptable for a HTTP request). Replacing the call to last_modified by a constant value (Ptime.epoch), the response time is reduced to sub-second.

I don't quite understand the implementation of `last_modified`, but is there a chance to use a different approach to speed it up by a factor of ~300? In the meantime, I can as well revert to not use irmin's last_modified, but store a last_modified timestamp as data.

irmin-last-modified

### PR DESCRIPTION
**Pull request by pascutto**
Fixes #933 (?)

### CODE REVIEW

**Comment by samoht** [comment]
Is this ready to merge? Could you rebase it to master (and fix the lint issue)?

Do you have some numbers to compare the performance with master?

**Comment by pascutto** [comment]
Rebased. 
We don't have a benchmark for this yet, and I'm not sure this actually improves the performance. Perhaps @hannesm you could test it with your data and see if that solves your issue?

**Comment by pascutto** [comment]
Just added a commit to use the latest version of `bheap`. That should improve the memory footprint, but probably won't make a difference for this issue.

File: CHANGES.md [hunk 1]
```diff
@@ -4,6 +4,9 @@
 
 - **irmin-pack***
   - Fix a major bug in the LRU which was never used (#1035, @samoht)
+  
+- **irmin***
+  - Improve `last_modified` performances (#948, @pascutto)
```
Comment (CraigFe) [inline CHANGES.md:9]:
```suggestion
  - Improve performance of `last_modified` (#948, @pascutto)
```

**Review by CraigFe — APPROVED** [review]
LGTM. Probably worth merging just to get rid of the `bheap` vendor. (Although I'm a bit sad that the library `bheap` introduces `Binary_heap` rather than `Bheap`...)

**Comment by CraigFe** [comment]
On second thought, we may want to wait until `bheap` uses Dune before introducing more issues for `duniverse`... (Tracked in a PR by @avsm: https://github.com/backtracking/bheap/pull/2)

**Comment by hannesm** [comment]
@pascutto sorry, I fail to find time to see whether this improvement (which it clearly is) is sufficient for my use. I'm a bit stuck with a different (memory-usage) error of git and irmin. to me, this PR improves last modified and can be merged in any case.

**Pull request merged**